craft_application/launchpad/models/distro.py

```diff
@@ -0,0 +1,74 @@
+#  This file is part of craft-application.
+#
+#  Copyright 2024 Canonical Ltd.
+#
+#  This program is free software: you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License version 3, as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful, but WITHOUT
+#  ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+#  SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Launchpad distro_series."""
+import enum
+
+# This file relies heavily on dynamic features from launchpadlib that cause pyright
+# to complain a lot. As such, we're disabling several pyright checkers for this file
+# since in this case they generate more noise than utility.
+# pyright: reportUnknownMemberType=false
+# pyright: reportUnknownVariableType=false
+# pyright: reportUnknownArgumentType=false
+# pyright: reportOptionalMemberAccess=false
+# pyright: reportAttributeAccessIssue=false
+# pyright: reportOptionalCall=false
+# pyright: reportOptionalIterable=false
+# pyright: reportOptionalSubscript=false
+# pyright: reportIndexIssue=false
+from .base import LaunchpadObject
+
+
+class DistributionTypes(enum.Enum):
+    """Types of distribution in Launchpad."""
+
+    DISTRIBUTION = "distribution"
```

Comment:
Curious that there's a type of distribution called "distribution". What are some other types?

Reply:
That's the only type, but I designed the `LaunchpadObject` so a specific class could handle multiple Launchpad types.